.github/workflows/release.yml

```diff
@@ -1,34 +1,42 @@
 name: Release
 
 on:
-  release:
-    types:
-      - published
+  pull_request:
+    types: [closed]
+    branches:
+      - master
 
 jobs:
-  upload_release:
-    name: Upload release
-    runs-on: ubuntu-20.04
+  create_release:
+    name: Create release
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.merged == true && startsWith(github.head_ref, 'release/')
     steps:
-      - uses: actions/checkout@v2
-        # Include all history and tags, needed for building the right version
+      - name: Get GitHub App token
+        id: get_token
+        uses: tibdex/github-app-token@v1
         with:
-          fetch-depth: 0
-
-      - uses: actions/setup-python@v2
-        name: Install Python
+          app_id: ${{ secrets.PIPELINE_GITHUB_APP_ID }}
+          private_key: ${{ secrets.PIPELINE_GITHUB_APP_PRIVATE_KEY }}
+      - name: Create release
+        env:
+          RELEASE_BRANCH: ${{ github.head_ref }}
+          SHA: $${ github.event.pull_request.head.sha }}
         with:
-          python-version: "3.10"
-
-      - name: Install datadog_checks_dev
-        run: |
-          python -m pip install datadog_checks_dev[cli]
+          github-token: ${{ steps.get_token.outputs.token }}
+          script: |
+            const tagName = process.env.RELEASE_TAG.split("/")[1];
```

Comment:
Im not too familiar with how github actions work but is the `RELEASE_TAG` env var shared between the workflows somehow? Also why do we split it in this case?

Reply:
Missed the variable name, should be `RELEASE_BRANCH`.